cli/lsp/analysis.rs

```diff
@@ -948,6 +984,136 @@ fn prepend_whitespace(content: String, line_content: Option<String>) -> String {
   }
 }
 
+/// Get LSP range from the provided SWC start and end locations.
+fn get_range_from_loc(start: &Loc, end: &Loc) -> lsp::Range {
+  lsp::Range {
+    start: lsp::Position {
+      line: (start.line - 1) as u32,
+      character: (start.col_display + 1) as u32,
+    },
+    end: lsp::Position {
+      line: (end.line - 1) as u32,
+      character: (end.col_display - 1) as u32,
+    },
+  }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct TextRange {
+  pub range: lsp::Range,
+  pub text: String,
+}
+
+impl TextRange {
```

Comment:
Doing that without the source map becomes a painful exercise of dealing with the line index.  But I will think about how to deal with the positions.  The biggest problem is we can't hang to so the SourceMap because it is an Rc and anything we hold a reference to in the language_server.  On the other hand, the conversion to and from the line index is use when communicating to tsc.  I think we already have utility functions for converting from swc positions to LSP positions and ranges using the line index (which are different than tsc ones... long story).

I will take a look at it all once again and see if we can end up in a better place.